base/commands/migration/migration_stages.go

```diff
@@ -0,0 +1,316 @@
+						}
+						execErr = errors.New(errs)
+						break StatusReaderLoop
+					case StatusCanceled, StatusCanceling:
+						execErr = errors2.ErrUserCancelled
+						break StatusReaderLoop
+					}
+					q := fmt.Sprintf(`SELECT JSON_QUERY(this, '$.migrations[%d]') FROM %s WHERE __key= '%s'`, i, StatusMapName, migrationID)
+					res, err := ci.Client().SQL().Execute(ctx, q)
+					if err != nil {
+						execErr = err
+						break StatusReaderLoop
+					}
+					iter, err := res.Iterator()
+					if err != nil {
+						execErr = err
+						break StatusReaderLoop
+					}
+					if iter.HasNext() {
+						row, err := iter.Next()
+						if err != nil {
+							execErr = err
+							break StatusReaderLoop
+						}
+						rowStr, err := row.Get(0)
+						if err != nil {
+							execErr = err
+							break StatusReaderLoop
+						}
+						var m DSMigrationStatus
+						if err = json.Unmarshal(rowStr.(serialization.JSON), &m); err != nil {
+							execErr = err
+							break StatusReaderLoop
+						}
+						status.SetProgress(m.CompletionPercentage)
+						switch m.Status {
+						case StatusComplete:
+							return nil, nil
+						case StatusFailed:
+							return nil, stage.IgnoreError(errors.New(m.Error))
+						case StatusCanceled:
+							execErr = errors2.ErrUserCancelled
+							break StatusReaderLoop
+						}
+					}
+					time.Sleep(1 * time.Second)
+				}
+				if execErr != nil {
+					status.SetText(execErr.Error())
+				}
+				return nil, execErr
+			},
+		})
+	}
+	return stages, nil
+}
+
+func dataStructuresToBeMigrated(ctx context.Context, ec plug.ExecContext, migrationID string) ([]DataStructureInfo, error) {
+	var dss []DataStructureInfo
+	ci, err := ec.ClientInternal(ctx)
+	if err != nil {
+		return nil, err
+	}
+	q := fmt.Sprintf(`SELECT this FROM %s WHERE __key= '%s'`, StatusMapName, migrationID)
+	res, err := ci.Client().SQL().Execute(ctx, q)
+	if err != nil {
+		return nil, err
+	}
+	it, err := res.Iterator()
+	if err != nil {
+		return nil, err
+	}
+	if it.HasNext() { // single iteration is enough that we are reading single result for a single migration
+		row, err := it.Next()
+		if err != nil {
+			return nil, err
+		}
+		r, err := row.Get(0)
+		if err != nil {
+			return nil, err
+		}
+		var status OverallMigrationStatus
+		if err = json.Unmarshal(r.(serialization.JSON), &status); err != nil {
+			return nil, err
+		}
+		for _, m := range status.Migrations {
+			dss = append(dss, DataStructureInfo{
+				Name: m.Name,
+				Type: m.Type,
+			})
+		}
+	}
+	return dss, nil
+}
+
+func saveMemberLogs(ctx context.Context, ec plug.ExecContext, ci *hazelcast.ClientInternal, migrationID string) error {
+	for _, m := range ci.OrderedMembers() {
+		l, err := ci.Client().GetList(ctx, DebugLogsListPrefix+m.UUID.String())
+		if err != nil {
+			return err
+		}
+		logs, err := l.GetAll(ctx)
+		if err != nil {
+			return err
+		}
+		for _, line := range logs {
+			ec.Logger().Info(fmt.Sprintf("[%s_%s] %s", migrationID, m.UUID.String(), line.(string)))
```

Comment:
🤔 Not sure we need member UUID in every line. The member logs should already be including it. Why did you want to add migration id and member id to every line? Also, where do we save the member logs to? Do we save all member logs in the same file? Is it because you wanted to add these two info to every line?

Reply:
We see member logs under `CLC_HOME/logs`. We add all logs to the same file, as you guessed that is why we wanted to add these two info to every line.